packages/js/product-editor/src/blocks/generic/linked-product-list/edit.tsx

```diff
@@ -143,6 +197,15 @@ export function LinkedProductListBlockEdit( {
 		);
 
 		setLinkedProductIds( newLinkedProductIds );
+		setSearchValue( '' );
+		setIsSearching( true );
+		searchProducts( '', [ ...( linkedProductIds || [] ), productId ] )
+			.then( ( products ) => {
+				setSearchedProducts( products );
+			} )
+			.finally( () => {
+				setIsSearching( false );
+			} );
```

Comment:
We repeated these lines a few times. It would be great to have an abstraction here.

Reply:
I originally had this abstracted and Maikel had suggested moving it out of the component, but given its predominantly state changes it did cause this duplicate code.
I end up moving it to its own function again: https://github.com/woocommerce/woocommerce/pull/50128/commits/b3b851835779ae238ed537748a356229bfc4ec24